src/wasm/wat-parser-internal.h

```diff
@@ -585,6 +585,30 @@ std::optional<LexStrResult> str(std::string_view in) {
   return ctx.lexed();
 }
 
+// keyword ::= ( 'a' | ... | 'z' ) idchar* (if literal terminal in grammar)
+// reserved ::= idchar+
```

Comment:
What are "literal terminal" and "reserved"? My guess is the first is the reserved keywords in wasm text grammar like `module`, `type`, or `func`, and the latter is just any other random sequence of chars, is that correct? If so it's ironical that the latter's name is "reserved".

Reply:
Your guess is correct and yep, it is a little ironic. Here's where the spec talks about this: https://webassembly.github.io/spec/core/text/lexical.html#text-reserved.